### PR TITLE
Support `actor` object conforming to `Actor` protocol

### PR DESCRIPTION
**Pull request by treastrain**
# Issue
Part of #216
(https://github.com/uber/mockolo/issues/216#issuecomment-1454179318 's "2.")

# Description
Add feature generating mocks for that protocol that conform to the [`Actor`](https://developer.apple.com/documentation/swift/actor) automatically become actor.

For details: https://github.com/uber/mockolo/issues/216#issuecomment-1454179318

### CODE REVIEW

File: Tests/TestActorMocking/FixtureMockableActor.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import MockoloFramework
+
+let actorProtocol =
+"""
+/// \(String.mockAnnotation)
```
Comment (uhooi) [inline Tests/TestActorMocking/FixtureMockableActor.swift:5]: I would also like you to write the following test.

```swift
/// \(String.mockAnnotation)
protocol Foo: Bar {
    func foo(arg: String) async -> Result<String, Error>
    var bar: Int { get }
}

protocol Bar: Actor {
    func baz(arg: String) async -> Result<String, Error>
    var qux: Int { get }
}
```

**Review by uhooi — CHANGES_REQUESTED** [review]
Thx PR!!

File: Tests/TestActorMocking/MockActorTests.swift [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import Foundation
+
+final class MockActorTests: MockoloTestCase {
+
+    func testActorProtocol() {
+        verify(srcContent: actorProtocol,
+               dstContent: actorProtocolMock)
+    }
+}
```
[the remark this answers is not in the document]
Comment (treastrain) [reply Tests/TestActorMocking/MockActorTests.swift:9]: or use `XCTSkip`

**Comment by uhooi** [comment]
@treastrain 
The test is failing in Xcode 12.
Since Swift Concurrency (async/await, Actor etc) is available from Xcode 13, please skip the test in Xcode 12.

```
Test Suite 'MockActorTests' started at 2023-03-10 13:00:08.892
Test Case '-[MockoloTests.MockActorTests testActorProtocol]' started.
/Users/runner/work/mockolo/mockolo/Tests/MockoloTestCase.swift:154: error: -[MockoloTests.MockActorTests testActorProtocol] : XCTAssertTrue failed
Test Case '-[MockoloTests.MockActorTests testActorProtocol]' failed (1.854 seconds).
Test Suite 'MockActorTests' failed at 2023-03-10 13:00:10.746.
	 Executed 1 test, with 1 failure (0 unexpected) in 1.854 (1.854) seconds
```

ref: https://github.com/uber/mockolo/actions/runs/4376819642/jobs/7676406579#step:5:251

**Review by sidepelican — CHANGES_REQUESTED** [review]
I can say the same things with https://github.com/uber/mockolo/pull/221 . See it.

**Comment by acevif** [comment]
Hi,

I'm really interested in this feature and looking forward to its potential inclusion.
Thank you for your work on this!

**Comment by treastrain** [comment]
I'm sorry. I have been away for a long time since I made this pull request.

I will defer the rest of this to #266.

**Pull request closed without merging**